azkaban-web-server/src/main/java/azkaban/webapp/WebMetrics.java

```diff
@@ -42,13 +44,24 @@
 
   WebMetrics() {
     registry = MetricsManager.INSTANCE.getRegistry();
-    setupAllMetrics();
+    setupStaticMetrics();
   }
 
-  private void setupAllMetrics() {
+  private void setupStaticMetrics() {
     webGetCall = MetricsUtility.addMeter("Web-Get-Call-Meter", registry);
     webPostCall = MetricsUtility.addMeter("Web-Post-Call-Meter", registry);
-    MetricsUtility.addLongGauge("fetchLogLatency", logFetchLatency, registry);
+    MetricsUtility.addGauge("fetchLogLatency", registry, logFetchLatency::get);
+  }
+
+  public void addExecutorManagerMetrics(ExecutorManager executorManager) {
```

Comment:
I wonder if centralizing all the registration of metrics is a good idea. This class now has dependencies on many other components which makes testing a bit harder.

Your new appoach of registering jetty related metrics inside the Jetty related code seems to get around this issue.